### DISCUSSION TITLE
custom WebSocket Theia backend service issue with multi-tent...

### DISCUSSION THREAD

**Discussion by eli88p** [eclipse-theia/theia#8108] [General] [answered]
Hi guys,
I need help with some Theia backend service which connect to WS and get live stream data.
The service has one more WS connection with theia fronted.
now my problem is when i try to connect with multi sessions from one instance of Theia, the messages arrive in other session.
Is there any effective solution?
![diagram](https://user-images.githubusercontent.com/11608934/85993927-b6080780-b9ff-11ea-8aca-361673bbd635.png)

Thanks in advance
Eli.

**Comment by akosyakov** [comment, marked as answer]
> now my problem is when i try to connect with multi sessions from one instance of Theia, the messages arrive in other session.


So many web sockets connections :) Could you elaborate what do you mean by multi sessions? Theia has a session only on frontend, on backend you can use connection (session) scoping to instantiate services only within this connection. You should be able to have such service which establish then websocket connection to Net. On connection scoping: https://github.com/eclipse-theia/theia/blob/master/packages/core/src/node/messaging/connection-container-module.ts#L35-L73

**Reply by eli88p** [reply]
thanks @akosyakov 
i mean by multi sessions - 2 users open Theia fronted that runs in kube pod with one replica
my problem is comunicate between theia frontend session with theia backed
i need to be able to subscribe each user (fronted) to another topic but because i have one instance of theia backend i'm having problems
Thanks in advance

**Reply by akosyakov** [reply]
Yes, then you need connection scoping. Each user will get own instance of backend services. Please look how types referenced in the previous comment are used in this repo.
